lib/internal/modules/esm/translators.js

```diff
@@ -158,12 +162,27 @@ translators.set('wasm', async function(url) {
   }
 
   const imports =
-      WebAssembly.Module.imports(compiled).map(({ module }) => module);
+    WebAssembly.Module.imports(compiled).map(({ module }) => module);
   const exports = WebAssembly.Module.exports(compiled).map(({ name }) => name);
+  const usesWasi = wasiEnabled && imports.includes('wasi_unstable');
```

Comment:
I'm a little surprised by the semantics here of basing decisions of loading semantics on whether `'wasi_unstable'` is being imported. These loader semantics are not standards-track. I wonder if we should make this determination based on a custom section in the module instead. (e.g., what if the module just uses other modules that use `wasi_unstable`?)

Reply:
It looks like WASI is making less generically named exported entry functions (like `__wasi_unstable_reactor_start`), which I think are a much better indicator. This is really just a POC atm.